server/src/core/payment/transaction-service.ts

```diff
@@ -359,4 +359,58 @@ export class Transactions implements TransactionService {
   private sendingProvider(): PaymentProvider {
     return this.providers.getPreferredForSending();
   }
+
+  async generateDistributionReportDocs(): Promise<DistributionReport[]> {
+    try {
+      const results: DistributionReport[] = await this.collection.aggregate<DistributionReport>([
+        { 
+          $match: { 
+            type: 'distribution', 
+            status: 'success',
+            updatedAt: { $gt: new Date(new Date().getTime() - (1 * 24 * 3600 * 1000)) }
```

Comment:
I think we should keep track when the last distribution report ran and aggregate transactions which succeeded after that time, rather than counting 24hours. I.e. each report aggregates the transactions that completed after the last report was computed. What do you think? I think this way makes it more tolerant to situations where the report doesn't run exactly in 24hr intervals. Also, hardcoding the time just doesn't follow the practice we've been following so far.

Reply:
But this means that we'd need to keep track of when the last report ran. While at the same time we should ensure that no transaction is counted twice (in two separate reports). And we should also ensure we don't miss transactions.